teamcity/resource_build_config.go

```diff
@@ -126,6 +126,10 @@ func resourceBuildConfig() *schema.Resource {
 							Type:     schema.TypeString,
 							Optional: true,
 						},
+						"mode": {
+							Type:     schema.TypeString,
```

Comment:
You can add a `ValidateFunc` here in conjunction with the `validation.StringInSlice` to make sure this can only take one of the valid values.